### PR TITLE
3.0-canonical-k8s

### PR DESCRIPTION
**Pull request by resmaa**
**Release 3.0 
Canonical Micro Kubernetes**

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/931#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @resmaa. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Prerequisites
+
+* An Ubuntu Operating System
+
+
+* 20GB Storage
+
+
+* 4G Memory
+
```
Comment (truegilb) [inline content/docs/06-integrations/00-canonical-k8s.md:41]: The above "prerequisites" seem to come from https://microk8s.io/docs/getting-started. Let's make a reference to their site if reader wants more info.

Comment (resmaa) [reply content/docs/06-integrations/00-canonical-k8s.md:41]: in references, section added the site

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+1. [Create AWS Cloud Account](/clusters/public-cloud/eks#creatinganawscloudaccount) in Palette Console.
+
+2. [Create a Infrastructure or Add-on Cluster Profile](/cluster-profiles/task-define-profile) for AWS cloud.
+
+3. Add Micro Kubernetes as the Kubernetes distribution for the `Infrastructure Layer`.
+
+4. Launch your Palette [AWS Cluster](/clusters/public-cloud/aws#deployinganawscluster).
+
+5. Wait for the provisioning to get completed.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:69]:
```suggestion
5. Wait for provisioning to complete.
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+* ** 1.23.4**
+
+</Tabs.TabPane>
+
+</Tabs>
+
+# Quick Workflow
+
+1. [Create AWS Cloud Account](/clusters/public-cloud/eks#creatinganawscloudaccount) in Palette Console.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:61]:
```suggestion
1. [Create an AWS Cloud Account](/clusters/public-cloud/eks#creatinganawscloudaccount) in the Palette Console.
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+2. [Create a Infrastructure or Add-on Cluster Profile](/cluster-profiles/task-define-profile) for AWS cloud.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:63]:
```suggestion
2. [Create an Infrastructure or Add-on Cluster Profile](/cluster-profiles/task-define-profile) for AWS cloud.
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+3. Add Micro Kubernetes as the Kubernetes distribution for the `Infrastructure Layer`.
+
+4. Launch your Palette [AWS Cluster](/clusters/public-cloud/aws#deployinganawscluster).
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:67]:
```suggestion
4. Launch Palette [AWS Cluster](/clusters/public-cloud/aws#deployinganawscluster).
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import Tabs from 'shared/components/ui/Tabs';
+import WarningBox from 'shared/components/WarningBox';
+import InfoBox from 'shared/components/InfoBox';
+import PointsOfInterest from 'shared/components/common/PointOfInterest';
+import Tooltip from "shared/components/ui/Tooltip";
+
+# Canonical Micro Kubernetes
+
+
+MicroK8s is a scalable Kubernetes distribution providing core Kubernetes functionalities with optimal resource consumption.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:20]:
```suggestion
Micro Kubernetes is a scalable Kubernetes distribution that provides core Kubernetes functionalities with optimal resource consumption. 
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import InfoBox from 'shared/components/InfoBox';
+import PointsOfInterest from 'shared/components/common/PointOfInterest';
+import Tooltip from "shared/components/ui/Tooltip";
+
+# Canonical Micro Kubernetes
+
+
+MicroK8s is a scalable Kubernetes distribution providing core Kubernetes functionalities with optimal resource consumption. 
+
+With limited resource requirements, MicroK8s extends its support in:
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:22]:
```suggestion
With limited resource requirements, Micro Kubernetes extends its support with the following:
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+* Turning Kubernetes into a lightweight development tool.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:24]:
```suggestion
* Turns Kubernetes into a lightweight development tool.
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+
+* Making Kubernetes available for use in minimal environments such as GitHub CI.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:27]:
```suggestion
* Makes Kubernetes available for use in minimal environments such as GitHub continuous integration (CI).
```

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+
+* Adapting Kubernetes for small-appliance IoT applications.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:30]:
```suggestion
* Adaps Kubernetes for small-appliance internet of things (IoT) applications.
```

**Review by ritawatson — CHANGES_REQUESTED** [review]
Minor suggested changes.

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+With limited resource requirements, Micro Kubernetes extends its support with the following:
+
+* Turns Kubernetes into a lightweight development tool.
+
+
+* Makes Kubernetes available for use in minimal environments such as GitHub Continuous Integration (CI).
+
+
+* Adaps Kubernetes for small-appliance internet of things (IoT) applications.
```
Comment (ritawatson) [inline content/docs/06-integrations/00-canonical-k8s.md:30]:
```suggestion
* Adapts Kubernetes for small-appliance internet of things (IoT) applications.
```

**Review by ritawatson — CHANGES_REQUESTED** [review]
Typo, then it's done. :)

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+5. Wait for provisioning to complete.
+
+<InfoBox>
+Currently Palette supports Micro Kubernetes for AWS clusters only.
+</InfoBox>
+
+## References
+
+* https://microk8s.io/
```
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-canonical-k8s.md:77]:
```suggestion
- [Micro k8s](https://microk8s.io/)
```

Let's use markdown links whenever possible.

File: content/docs/06-integrations/00-canonical-k8s.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+
+* https://github.com/canonical/microk8s
```
Comment (karl-cardenas-coding) [inline content/docs/06-integrations/00-canonical-k8s.md:80]:
```suggestion
- [Canonical Micro k8s](https://github.com/canonical/microk8s)
```

**Review by karl-cardenas-coding — APPROVED** [review]
Minor changes, but overall good to go.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/931#pullrequestreview-1199947249" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/931#" title="Author self-approved">resmaa</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request closed without merging**